src/include/migraphx/op/prefix_scan_op.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_SCAN_OP_HPP
+#define MIGRAPHX_GUARD_OPERATORS_SCAN_OP_HPP
+
+#include <migraphx/op/name.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+template <class Derived>
+struct prefix_scan_op : op_name<Derived>
+{
+    int64_t axis;
+    bool exclusive = false;
+    bool reverse   = false;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(
+            f(self.axis, "axis"), f(self.exclusive, "exclusive"), f(self.reverse, "reverse"));
+    }
+
+    value attributes() const
+    {
+        value normalize;
+        normalize["axis"] = value::array{normalize_attribute::include_min};
+        return {{"normalize_axes", normalize}};
+    }
+
+    shape normalize_compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1);
```

Comment:
we may need a standard() check here since it looks like algorithm below assume this.

Reply:
If the input shape is broadcasted and result is assigned as ```argument result = args[0];``` as below, the result will not have enough buffer to store the results. 

For the transposed shape, I think we might need to tune the axis (I am not very sure, but it might be necessary)

Reply:
> If the input shape is broadcasted and result is assigned

We can probably add a `not_broadcasted` as I dont think it makes much sense to do a prefix sum on a broadcasted tensor.

> For the transposed shape, I think we might need to tune the axis (I am not very sure, but it might be necessary)

No we don't for the same reason we don't need to tune the axis for softmax or reduce.